ui/src/data_explorer/containers/DataExplorer.tsx

```diff
@@ -523,13 +523,23 @@ export class DataExplorer extends PureComponent<Props, State> {
       this.templates
     )
 
-    const queryDraft = {
-      query,
-      queryConfig,
-      source: source.links.self,
-      type: QueryType.InfluxQL,
+    if (this.service) {
+      const queryDraft = {
+        query,
+        queryConfig,
+        source: sourceLink,
+        type: QueryType.Flux,
+      }
+      loadDE([queryDraft], timeRange)
```

Comment:
could this line could be factored out to after the conditional? seems duplicated

Reply:
Does not save an LOC because we would have to `let` queryDraft above the conditional and seems more direct the way it is.

Reply:
ah yes